test/extensions/filters/http/common/compressor/compressor_integration_tests.cc

```diff
@@ -0,0 +1,297 @@
+#include "test/extensions/filters/http/common/compressor/compressor_integration_tests.h"
```

Comment:
I would love a high level explanation of what we're testing in this file. I think it's that upgrades work through the compressor filter, but its not clear without reading a lot of code what kind of interactions (if any?) we're trying to validate

Reply:
ok. will add